django-stubs/db/backends/mysql/base.pyi

```diff
@@ -39,20 +40,20 @@ class DatabaseWrapper(BaseDatabaseWrapper):
     validation_class: type[DatabaseValidation]
 
     vendor: str
-    data_types: Any
-    operators: Any
+    data_types: dict[str, str]
+    operators: dict[str, str]
     pattern_esc: str
-    pattern_ops: Any
-    isolation_levels: Any
+    pattern_ops: dict[str, str]
+    isolation_levels: set[str]
     Database: Any
```

Comment:
It is `Database = Database` https://github.com/django/django/blob/9a27c76021f934201cccf12215514a3091325ec8/django/db/backends/mysql/base.py#L202

Reply:
For external libraries, not `django`, could you tell me how to check the type?

Reply:
You can also define a custom `Protocol` class that lists all members expected to exist in the database module. Mypy at least supports assigning modules to a Protocol.

Reply:
Can you tell me what is the best file to place a `Protocol` custom class?

Reply:
near the place where they are used :)